pkg/genericapiserver/genericapiserver.go

```diff
@@ -590,10 +591,12 @@ func (s *GenericAPIServer) Run(options *ServerRunOptions) {
 			alternateDNS := []string{"kubernetes.default.svc", "kubernetes.default", "kubernetes"}
 			// It would be nice to set a fqdn subject alt name, but only the kubelets know, the apiserver is clueless
 			// alternateDNS = append(alternateDNS, "kubernetes.default.svc.CLUSTER.DNS.NAME")
-			if err := util.GenerateSelfSignedCert(s.ClusterIP.String(), options.TLSCertFile, options.TLSPrivateKeyFile, alternateIPs, alternateDNS); err != nil {
-				glog.Errorf("Unable to generate self signed cert: %v", err)
-			} else {
-				glog.Infof("Using self-signed cert (%options, %options)", options.TLSCertFile, options.TLSPrivateKeyFile)
+			if !fileExists(options.TLSCertFile) || !fileExists(options.TLSPrivateKeyFile) {
```

Comment:
I suggest:

* If only one of these files is present, or readable, do a fatal error-- that's probably a typo or something. The user should give us both or neither. (And what about the CA file? I guess we require that to be stuffed in with the cert?)